src/o365/teams/commands/team/team-set.ts

```diff
@@ -1,26 +1,23 @@
+import * as fs from 'fs';
+import * as path from 'path';
 import config from '../../../../config';
 import commands from '../../commands';
-import GlobalOptions from '../../../../GlobalOptions';
 import {
   CommandOption, CommandValidate
 } from '../../../../Command';
 import Utils from '../../../../Utils';
-import request from '../../../../request';
 import GraphCommand from '../../../base/GraphCommand';
+import { GroupUpdateService, Options} from '../../../aad/services/GroupUpdateService';
 
 const vorpal: Vorpal = require('../../../../vorpal-init');
 
-interface CommandArgs {
-  options: Options;
-}
-
-interface Options extends GlobalOptions {
+interface TeamSetOptions extends Options{
   teamId: string;
-  displayName?: string;
-  description?: string;
-  mailNickName?: string;
-  classification?: string;
-  visibility?: string;
+  visibility?:string;
+  imagePath?: string;
```

Comment:
I can do that. If there should be a difference, the Teams option should rather be 'picture' instead of image, that's how it is expressed when changing in the UI anyway.

If I align id and logoPath option names, isPrivate/visibility is the only discrepancy left. Would changing the o365 group option from bool to string and renaming it 'visibility' be too much (breaking change)?

Reply:
Let's align id and logoPath. Since UI tends to change, it could be hard to keep it solid over time, so perhaps using a more technical term is better.

Renaming an option would be a breaking change. We could introduce an alias for the time-being and then align them in the next major version but I wonder if there are any downsides to it given the difference in the API. Seems like a string-based visibility is leaving room for future adjustments whereas a boolean is fixed. Going either way could lead us to a dead-end where an additional visibility options would get introduced in the future and then we'd either have to expose it on both objects or we'd have another breaking change. Tough call, but perhaps better to adhere to what's available in the API.

Reply:
There never is a bool in the API for representing visibility (there is only the Group endpoint for setting these properties, whether the group is teamsyfied or not). 

I think it stems from a design choice to map the graph request string representation of visibility to a bool called isPublic in OfficeDevPnp.Core. From there it's been used by the PnpPowerShell (Set-PnpUnifiedGroup) and, maybe, then for the sake of parity/alignment the cli command also uses the isPublic, mapping a bool to a string.

The notion of the group visibility scope being a binary state I think is an interpretation of a string that (as you point out) up until now has had two valid values.

It would be interesting to see if the PnpPowerShell, for the sake of convenience, decides to implement an Update-PnpTeam cmdlet. Would they then choose to stick with the isPublic worldview or go with visibility and map the string to a bool which is then mapped from bool to string in the OfficeDevPnp.Core. :)

Reply:
Convenience is definitely helpful for daily use, but it should align with the intent expressed by the API and at times it's hard if the intent is not documented. If we based our original choice on what's in PnP PowerShell and that doesn't seem to be in line with the API, then perhaps we should consider aligning our command with the API and in CLI v2 having two options that do the same which we could then clean up in the next major version v3.

Reply:
I just saw this whel looking through the API docs (as you do) 

visibility | String | Specifies the visibility of an Office 365 group. Possible values are: Private, Public, or Hiddenmembership; blank values are treated as public. See group visibility options to learn more.Visibility can be set only when a group is created; it is not editable.Visibility is supported only for unified groups; it is not supported for security groups.Returned by default.
-- | -- | --

From https://docs.microsoft.com/en-us/graph/api/resources/group?view=graph-rest-1.0

So it seems that there are three values `public, private, hiddenmembership`.

Reply:
Good catch. Let's align both commands on this and keep the existing boolean as a shorthand for public/private. We can make a separate note to remove it in the next major version.